pkg/kv/kvserver/store_rebalancer.go

```diff
@@ -379,24 +378,6 @@ func (sr *StoreRebalancer) chooseLeaseToTransfer(
 	storeMap map[roachpb.StoreID]*roachpb.StoreDescriptor,
 	options *allocatorimpl.QPSScorerOptions,
 ) (replicaWithStats, roachpb.ReplicaDescriptor, []replicaWithStats) {
-	// NB: Don't switch over to the new locality-aware lease transfer scheme until
-	// the cluster version is finalized.
-	if !sr.st.Version.IsActive(ctx, clusterversion.EnableNewStoreRebalancer) {
-		log.Infof(
-			ctx, "cluster version has not been finalized; using pre-22.1 load-based lease transfer scheme",
-		)
-
-		// We manually compute the cluster level under/over-fullness thresholds
-		// since the deprecated rebalance logic doesn't care about equivalence
-		// classes.
-		qpsMinThreshold := allocatorimpl.UnderfullQPSThreshold(options, allStoresList.CandidateQueriesPerSecond.Mean)
-		qpsMaxThreshold := allocatorimpl.OverfullQPSThreshold(options, allStoresList.CandidateQueriesPerSecond.Mean)
-		return sr.deprecatedChooseLeaseToTransfer(
```

Comment:
I believe there are no other usages of deprecatedChooseLeaseToTransfer, @aayushshah15 added it in #80190 to version gate the new 22.1 behaviour by keeping the 21.2 one around, but can be removed in this PR. Likewise for deprecatedChooseRangeToRebalance.

Reply:
The EnableNewStoreRebalancer also LGTM otherwise.

Reply:
Sent a PR: #86060.